packages/ecs-agent/Cargo.toml

```diff
@@ -13,10 +13,17 @@ path = "pkg.rs"
 url = "https://github.com/aws/amazon-ecs-agent/archive/v1.50.2/amazon-ecs-agent-v1.50.2.tar.gz"
 sha512 = "5bc946a529c2678b0cd45c3be556b116c26f6d3a3fe3f1cd598cf4fc09eaa8de1ed514bbf62b78892339980b647be49b5152508219181fe567b01152d230db31"
 
-# TODO: Package the CNI plugins
 # The ECS agent repository includes two CNI plugins as git submodules.  git
 # archive does not include submodules, so the tarball above does not include
-# the source of those plugins.
+# the source of those plugins.  Instead, we include the CNI plugin source
+# directly.
+[[package.metadata.build-package.external-files]]
+url = "https://github.com/aws/amazon-ecs-cni-plugins/archive/55b2ae77ee0bf22321b14f2d4ebbcc04f77322e1/amazon-ecs-cni-plugins.tar.gz"
+sha512 = "42e2dad0741f02a3b0d675c0f512557a118cef6df7e9d989a35c73d02484912d800453148abbf6d8424c5e40d2662745a9aef4a68cb2f610d9defc976e394f1f"
+
+[[package.metadata.build-package.external-files]]
+url = "https://github.com/aws/amazon-vpc-cni-plugins/archive/a21d3a41f922e14c19387713df66be3e4ee1e1f6/amazon-vpc-cni-plugins.tar.gz"
+sha512 = "86011a1d7a8d4a2fe824c9eea6ad2ef9394059475ffa421c3636afd4f4031d60898992022e34ce2b1e9594fa79766f747a6c239e26be5d71fe96addc46bf7f54"
```

Comment:
Is it possible to break the ecs-agent and the CNI plugins into two packages? Are there cross dependencies in the code itself?

I'd prefer to have a simpler build of the two where patches and such were well scoped to the agent and CNI plugins over having them combined and do a dance to build them together in the same package.

Reply:
I thought about this a lot as I was working on this.  While the CNI plugins exist in a separate git repository, the reality of how they're used in ECS is that they're incredibly tightly-coupled; there is no support for running versions of the CNI plugins that are out of sync with the versions expected to be used by the ECS agent (controlled via git submodules in the ECS agent source repository).  While the overall spec is more complicated with the patching work, the approach here makes it very obvious when one changes without the other and my hope is that it will be an overall improvement in our ability to effectively maintain the ECS agent + the plugins over time.